### PR TITLE
CD for Website      

### PR DESCRIPTION
**Pull request by jahn18**
### Description
This Github workflow will automatically deploy the website to a specified VM.

#### Setup Steps:   
1. Create a remote VM using some cloud provider. 
2. Add an SSH key to the VM (e.g. public SSH key). 
4. Ensure Docker is installed on VM 
	* See resources below.
5. Update the three secrets `SSH_HOST`, `SSH_USERNAME`, and `SSH_KEY` on Github 

Once these four steps are done, the deployment should work successfully!  

### Verification

I tested this workflow using a personal VM from DigitalOcean. 

Some scenarios:
- Test on VM that did not have `sailbot_workspace` cloned ✅ 
- Test on VM that already had `sailbot_workspace` cloned ✅ 
- Pushed small UI changes to the website to see if it successfully updates it on VM ✅ 

### Resources

- https://docs.docker.com/engine/install/ubuntu/

### CODE REVIEW

**Review by patrick-5546 — COMMENTED** [review]
Should be reviewed by the people who will be using this. Perhaps documentation could be in a more permanent place? But yea what do you think future leads?

File: .github/workflows/deployment.yml
Comment (SPDonaghy) [inline .github/workflows/deployment.yml:5]: Will the website go down briefly, while the container rebuilds, each time there is a push to main?

**Review by SPDonaghy — COMMENTED** [review]
Nice, yeah this is great. It would be good to just add the steps in this PR to somewhere in the docs.

File: .github/workflows/deployment.yml
Comment (jahn18) [reply .github/workflows/deployment.yml:5]: Yeah that's right, the website will go down as it's rebuilds, but the DB is persisted so it should be okay (we won't lose any of the data and the UI will start up back up where it left off). 

I think we can go with this for now, the downtime shouldn't be too bad. If we really wanted to ensure the website never goes down there are some deployment strategies such as [blue/green and red/black](https://octopus.com/blog/blue-green-red-black ), but imo it might be overkill. 

Might be worth considering in the future though!

**Comment by jahn18** [comment]
Thanks @SPDonaghy 😄 Apologizes for the late response, I just started work so things got a bit busy. I'm going to hand off the remaining work to @JordanChen123 because it'll be important for him to familarize himself with this. 

I added the steps in the PR to the docs here: https://ubcsailbot.atlassian.net/wiki/spaces/prjt22/pages/2048032924/Website+Deployment

File: .github/workflows/deployment.yml
Comment (SPDonaghy) [reply .github/workflows/deployment.yml:5]: Sounds good, yeah I don't see it being much of an issue, good to know there's some solutions if it does become a problem. I sure hope the website pops off! 😆

File: .github/workflows/deployment.yml
Comment (jahn18) [reply .github/workflows/deployment.yml:16]: Hey @SPDonaghy, I want to get this PR merged in because it's blocking another PR I opened up: https://github.com/UBCSailbot/sailbot_workspace/pull/386

If everything else looks good to you, would you mind approving this PR? I created a new issue for this task https://github.com/UBCSailbot/sailbot_workspace/issues/387, so @JordanChen123 can work on it separately.

Comment (SPDonaghy) [reply .github/workflows/deployment.yml:16]: Yeah of course!

**Review by SPDonaghy — APPROVED** [review]

**Comment by jahn18** [comment]
Once I get your approval @patrick-5546 should be good to go :)

**Review by patrick-5546 — APPROVED** [review]

**Pull request merged**
